### PR TITLE
[Merged by Bors] - feat: port Data.Prod.Basic

### PR DESCRIPTION
**Pull request by pechersky**
moves and restructures existing Data.Prod

### CODE REVIEW

File: Mathlib/Data/Prod/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+theorem mk.inj_right {α β : Type _} (b : β) :
+    Function.Injective (fun a => Prod.mk a b : α → α × β) := by
+  intro b₁ b₂ h
+  simpa only [and_true, eq_self_iff_true, mk.inj_iff] using h
+
+theorem ext_iff {p q : α × β} : p = q ↔ p.1 = q.1 ∧ p.2 = q.2 := by
+  rw [← @mk.eta _ _ p, ← @mk.eta _ _ q, mk.inj_iff]
+
+@[ext]
+theorem extₓ {α β} {p q : α × β} (h₁ : p.1 = q.1) (h₂ : p.2 = q.2) : p = q :=
```
Comment (semorrison) [inline Mathlib/Data/Prod/Basic.lean:98]: I don't think we should be using `ₓ` in names! This is/was just an internal mathport thing.

Comment (pechersky) [reply Mathlib/Data/Prod/Basic.lean:98]: Would you prefer the previous `ext'` name?

File: scripts/lint-style.py [hunk 2]
```diff
@@ -204,7 +204,7 @@ def regular_check(lines, path):
         if copy_done and line == "\n":
             continue
         words = line.split()
-        if words[0] != "import" and words[0] != "/-!":
+        if words != ["prelude"] and words[0] != "import" and words[0] != "/-!":
```
Comment (semorrison) [inline scripts/lint-style.py:207]: Let's revert this.

File: Mathlib/Data/Prod/Basic.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Data/Prod/Basic.lean:98]: Yes.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**